src/Stack/Upload.hs

```diff
@@ -203,13 +205,14 @@ mkUploader config us = do
         { upload_ = \tarName bytes -> do
             let formData = [partFileRequestBody "package" tarName (RequestBodyLBS bytes)]
             req2 <- formDataBody formData req1
-            let req3 = applyBasicAuth
+            mreq3 <- applyDigestAuth
                     (encodeUtf8 $ hcUsername creds)
                     (encodeUtf8 $ hcPassword creds)
                     req2
+                    manager
             putStr $ "Uploading " ++ tarName ++ "... "
             hFlush stdout
-            withResponse req3 manager $ \res ->
+            withResponse (fromMaybe req2 mreq3) manager $ \res ->
```

Comment:
I see that; my point is, I'd expect the error would be misleading. If anything hinted at "no password provided", I'd wonder why stack isn't providing a password, not why Hackage is rejecting digest authentication. I'm not implying some hypothetical user would be confused, I'm implying *I* would be confused.